### PR TITLE
Feature/data redis

### PR DESCRIPTION
**Pull request by guttff**
Data Redis Module

## Description 

Create a data redis module
- base redis module
- sample dat module book example

### CODE REVIEW

File: data/data-samples/sample-data-redis-book/pom.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            <groupId>it.ozimov</groupId>
+            <artifactId>embedded-redis</artifactId>
+            <version>0.7.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>1.17.5</version>
+            <scope>test</scope>
```
Comment (whoswendy) [inline data/data-samples/sample-data-redis-book/pom.xml]: remove the dep version and add deps to main pom.xml

File: data/data-samples/sample-data-redis-book/src/main/resources/data-book-application.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+spring.redis.database=0
```
Comment (whoswendy) [inline data/data-samples/sample-data-redis-book/src/main/resources/data-book-application.properties:14]: add copyright

File: data/data-samples/sample-data-redis-book/src/test/java/io/americanexpress/data/book/config/BookDataConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package io.americanexpress.data.book.config;
```
Comment (whoswendy) [inline data/data-samples/sample-data-redis-book/src/test/java/io/americanexpress/data/book/config/BookDataConfigTest.java:14]: add copyright

File: data/data-samples/sample-data-redis-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package io.americanexpress.data.book.config;
```
Comment (whoswendy) [inline data/data-samples/sample-data-redis-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java:14]: add copyright

File: data/data-samples/sample-data-redis-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
+package io.americanexpress.data.book.entity;
```
Comment (whoswendy) [inline data/data-samples/sample-data-redis-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java:14]: add copyright

File: data/data-samples/sample-data-redis-book/src/test/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,2 @@
+spring.redis.host=localhost
```
Comment (whoswendy) [inline data/data-samples/sample-data-redis-book/src/test/resources/application.properties]: add copyright

File: data/synapse-data-redis/pom.xml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            <artifactId>jedis</artifactId>
+            <version>3.3.0</version>
+            <type>jar</type>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-redis</artifactId>
+            <version>2.7.2</version>
+        </dependency>
+<!--        <dependency>-->
```
Comment (whoswendy) [inline data/synapse-data-redis/pom.xml]: remove commented code

File: data/synapse-data-redis/pom.xml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        <artifactId>data</artifactId>
+        <version>0.3.8-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>synapse-data-redis</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.data</groupId>
```
Comment (whoswendy) [inline data/synapse-data-redis/pom.xml:32]: remove version from deps and add deps to main pom

File: data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseReactiveRedisDataConfig.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package io.americanexpress.synapse.data.redis.config;
```
Comment (whoswendy) [inline data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseReactiveRedisDataConfig.java]: add copyright

File: data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseRedisDataConfig.java [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+    @Bean
+    JedisConnectionFactory jedisConnectionFactory() {
+        JedisConnectionFactory jedisConFactory = new JedisConnectionFactory();
+        jedisConFactory.setHostName("localhost");
+        jedisConFactory.setPort(6379);
+        jedisConFactory.afterPropertiesSet();
+        return jedisConFactory;
+    }
+//    @Bean
+//    JedisConnectionFactory jedisConnectionFactory() {
```
Comment (whoswendy) [inline data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseRedisDataConfig.java]: remove the commented code?

File: data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseRedisDataConfig.java [hunk 11]
```diff
@@ -0,0 +1,110 @@
+package io.americanexpress.synapse.data.redis.config;
```
Comment (whoswendy) [inline data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseRedisDataConfig.java:14]: add copyright

File: data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/entity/BaseEntity.java [hunk 12]
```diff
@@ -0,0 +1,98 @@
+package io.americanexpress.synapse.data.redis.entity;
```
Comment (whoswendy) [inline data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/entity/BaseEntity.java:14]: add copyright

File: data/data-samples/sample-data-redis-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java (hunk 4 above)
Comment (guttff) [reply data/data-samples/sample-data-redis-book/src/main/java/io/americanexpress/data/book/config/BookDataConfig.java:14]: done

File: data/data-samples/sample-data-redis-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java (hunk 5 above)
Comment (guttff) [reply data/data-samples/sample-data-redis-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java:14]: done

File: data/data-samples/sample-data-redis-book/src/main/resources/data-book-application.properties (hunk 2 above)
Comment (guttff) [reply data/data-samples/sample-data-redis-book/src/main/resources/data-book-application.properties:14]: done

File: data/data-samples/sample-data-redis-book/src/test/java/io/americanexpress/data/book/config/BookDataConfigTest.java (hunk 3 above)
Comment (guttff) [reply data/data-samples/sample-data-redis-book/src/test/java/io/americanexpress/data/book/config/BookDataConfigTest.java:14]: done

File: data/synapse-data-redis/pom.xml (hunk 8 above)
Comment (guttff) [reply data/synapse-data-redis/pom.xml:32]: done

File: data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseRedisDataConfig.java (hunk 11 above)
Comment (guttff) [reply data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseRedisDataConfig.java:14]: done

File: data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseRedisDataConfig.java (hunk 10 above)
Comment (guttff) [reply data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseRedisDataConfig.java]: done

File: data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/entity/BaseEntity.java (hunk 12 above)
Comment (guttff) [reply data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/entity/BaseEntity.java:14]: done

File: data/data-samples/sample-data-redis-book/pom.xml (hunk 1 above)
Comment (guttff) [reply data/data-samples/sample-data-redis-book/pom.xml]: done

File: data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseReactiveRedisDataConfig.java (hunk 9 above)
Comment (guttff) [reply data/synapse-data-redis/src/main/java/io/americanexpress/synapse/data/redis/config/BaseReactiveRedisDataConfig.java]: done

**Review by whoswendy — APPROVED** [review]

**Pull request merged**